plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -59,22 +63,98 @@ local OPDSBrowser = Menu:extend{
 function OPDSBrowser:init()
     self.item_table = self:genItemTableFromRoot()
     self.catalog_title = nil
-    self.title_bar_left_icon = "plus"
+    self.title_bar_left_icon = "appbar.menu"
     self.onLeftButtonTap = function()
-        self:addEditCatalog()
+        self:showOPDSMenu()
     end
     Menu.init(self) -- call parent's init()
 end
 
+function OPDSBrowser:showOPDSMenu()
+    local dialog
+    dialog = ButtonDialog:new{
+        buttons = {
+            {{
+                    text = _("Add catalog"),
+                    callback = function()
+                        UIManager:close(dialog)
+                        self:addEditCatalog()
+                    end,
+                    align = "left",
+            }},
+            {},
+            {{
+                    text = _("Sync all catalogs"),
+                    callback = function()
+                        UIManager:close(dialog)
+                        NetworkMgr:runWhenConnected(function()
+                            self.sync = true
```

Comment:
self.sync is always true when calling checkSyncDownload().
You can set it to true right in checkSyncDownload(), to avoid repeating 4 times in the callbacks.